feat(data): add API record id to Storage file prefix for API + Storage docs; update comments; update callout (#5533)
aws-amplify/docs: src/fragments/lib/graphqlapi/js/working-with-files.mdx

## Create a record with an associated file

> First create the API record, then upload the file to Storage, and finally add the association between the record and file. Use the following example with the GraphQL API and Storage categories to create a record and associate the file with the record.

Reviewer: ```suggestion
First create a record via the GraphQL API, then upload the file to Storage, and finally add the association between the record and file. Use the following example with the GraphQL API and Storage categories to create a record and associate the file with the record.
```

<Callout>

> The API record's `id` is prepended to the Storage file name to ensure uniqueness. If this is excluded, multiple API records could then be associated with the same file key unintentionally.

Reviewer: ```suggestion
Prepend a record's `id` to the Storage file name to ensure uniqueness. If this is excluded, multiple records could then be associated with the same file key unintentionally.
```

</Callout>

<BlockSwitcher>
<Block name="TypeScript">

```ts
const createSongDetails: CreateSongInput = {
  name: `My first song`,
};

// Create the API record:
const response = await API.graphql<GraphQLQuery<CreateSongMutation>>({
  query: mutations.createSong,
  variables: { input: createSongDetails },
});

> const _song = response?.data?.createSong;

Reviewer 2: Why is this variable prepended with an underscore? Is there something special about it vs all the others?

Author: I'd previously been using this naming convention, and later switched. Will make this consistent with the other variables.

Author: This is not needed, will update.

if (!_song) return;

// Upload the Storage file:
const result = await Storage.put(`${_song.id}-${file.name}`, file, {
  contentType: "image/png", // contentType is optional
});

const updateSongDetails: UpdateSongInput = {
  id: _song.id,
  coverArtKey: result?.key,
};

// Add the file association to the record:
const updateResponse = await API.graphql<
  GraphQLQuery<UpdateSongMutation>
>({
  query: mutations.updateSong,
  variables: { input: updateSongDetails },
});

const updatedSong = updateResponse?.data?.updateSong;

> // Check that the record has an associated image:

Reviewer 2: Nit
```suggestion
// Ensure the record has an associated image:
```

        id: _song.id,
      };

>       // const deletedSong = await API.graphql<GraphQLQuery<DeleteSongMutation>>({

Reviewer 2: Is this line needed?

Author: Nope, will remove.
